components/buttongroup/index.css

```diff
@@ -11,7 +11,28 @@ governing permissions and limitations under the License.
 */
 
 .spectrum-ButtonGroup {
-  --spectrum-buttongroup-button-gap-reset: 0;
+  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-300);
+  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-300);
+}
+
+.spectrum-ButtonGroup--sizeS {
+  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-200);
+  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-200);
+}
+
+.spectrum-ButtonGroup--sizeM {
+  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-300);
+  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-300);
+}
+
+.spectrum-ButtonGroup--sizeL {
+  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-300);
+  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-300);
+}
+
+.spectrum-ButtonGroup--sizeXL {
+  --spectrum-buttongroup-spacing-horizontal: var(--spectrum-spacing-300);
+  --spectrum-buttongroup-spacing-vertical: var(--spectrum-spacing-300);
 }
 
 .spectrum-ButtonGroup {
```

Comment:
Line 40 - I think we might need **flex-wrap: wrap** here because worse case scenerio could have horizontal overflow:
<img width="536" alt="Screen Shot 2022-09-27 at 4 33 46 PM" src="https://user-images.githubusercontent.com/25614178/192629770-7abb5408-669c-492f-868e-61b094ce124e.png">

Reply:
It looks like I will end up with something similar to the Vertical variant if I use `flex-wrap` and update the `inline` and `block` margins.

What do you think about using vertical variant styles as the default and applying the horizontal styles at this breakpoint instead?

<img width="417" alt="Screen Shot 2022-09-27 at 4 05 43 PM" src="https://user-images.githubusercontent.com/16216104/192635358-03b1cc6c-219e-4345-a752-f62c736ec0dd.png">

Reply:
Discussed with Monet and I think I was overcomplicating it thinking about layout changes.

It looks like I just need to use `gap` with `flex-wrap`:

<img width="430" alt="Screen Shot 2022-09-28 at 9 52 43 AM" src="https://user-images.githubusercontent.com/16216104/192812019-92b2d662-1f69-4dbe-b0f0-30a2f00ae09f.png">